core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -251,6 +251,14 @@ private[spark] object JettyUtils extends Logging {
   private def attachPrefix(basePath: String, relativePath: String): String = {
     if (basePath == "") relativePath else (basePath + relativePath).stripSuffix("/")
   }
+
+  /**
+   * Attach a prefix to the given path, but avoid returning an empty path.
+   * Don't remove the trailing / if present.
+   */
+  private def attachPrefixForRedirect(basePath: String, relativePath: String): String = {
+    if (basePath == "") relativePath else (basePath + relativePath)
```

Comment:
In fact, the conditional makes sense in the original method, because of the `stripSuffix` call, but does not make sense here - nor does the copied scaladoc about "avoid returning an empty path".